# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/MediaInfo.h
# Commit: eed72505cf9b
# Full Hash: eed72505cf9b3c968aced5a6488f2660e1f02da4
# Author: Alex Chronopoulos <achronop@gmail.com>
# Date: 2019-08-06 16:15:05
# Regressor Bug: 1530996
# File Overlap Count: 1
# Description:
#   Bug 1530996 - Use the new storage mechanism to save the current playback benchmark. r=jya
#   
#   Create an event in MediaFormatReader the will signal to the HTMLMediaElement to initiate a new storing according to the latest VideoInfo. Also when the application is shutting down, trigger a new storing early enough, before all the events are disconnected.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D38316
# ==============================================================================

diff -r 426b558f2f43 -r eed72505cf9b dom/media/MediaInfo.h
--- a/dom/media/MediaInfo.h	Tue Aug 06 09:24:40 2019 +0000
+++ b/dom/media/MediaInfo.h	Tue Aug 06 09:25:45 2019 +0000
@@ -253,6 +253,9 @@
   // bits etc)
   gfx::ColorRange mColorRange = gfx::ColorRange::LIMITED;
 
+  Maybe<int32_t> GetFrameRate() const { return mFrameRate; }
+  void SetFrameRate(int32_t aRate) { mFrameRate = Some(aRate); }
+
  private:
   // mImage may be cropped; currently only used with the WebM container.
   // A negative width or height indicate that no cropping is to occur.
@@ -260,6 +263,8 @@
 
   // Indicates whether or not frames may contain alpha information.
   bool mAlphaPresent = false;
+
+  Maybe<int32_t> mFrameRate;
 };
 
 class AudioInfo : public TrackInfo {